daemon/api_validate.go

```diff
@@ -414,3 +422,27 @@ func getSingleSeqFormingAssertion(st *state.State, accountID, name string, seque
 
 	return as, nil
 }
+
+func enforceValidationSet(st *state.State, accountID, name string, sequence, userID int) Response {
+	snaps, err := installedSnaps(st)
+	if err != nil {
+		return InternalError(err.Error())
+	}
+	vs, err := validationSetAssertionForEnforce(st, accountID, name, sequence, userID, snaps)
+	if err != nil {
+		// XXX: provide more specific error kinds? This would probably require
+		// assertstate.ValidationSetAssertionForEnforce tuning too.
+		return BadRequest("cannot enforce validation set: %v", err)
```

Comment:
in the spread test we will want to check for an error case too, and make sure the final error the user sees is not too repetitive

Reply:
It's not ideal:
```
# snap validate --enforce xSfWKGdLoQBoQx88vIM1MpbFNMq53t1f/bar=1
error: cannot apply validation set: cannot enforce validation set: validation sets assertions are not met:
- missing required snaps:
  - hello-world (required by sets xSfWKGdLoQBoQx88vIM1MpbFNMq53t1f/bar)
```

but I think making it better needs a few changes in other places and is worth a PR on its own?